From ba317cd9cef1fda959b3a7c56aa8e500f36d0fb4 Mon Sep 17 00:00:00 2001
From: Nathan Chancellor <natechancellor@gmail.com>
Date: Wed, 3 Oct 2018 14:39:42 -0700
Subject: [PATCH 02/11] DO-NOT-UPSTREAM: Hide -Wconstant-conversion behind W=3

There are a few of these warnings sprinkled throughout the kernel and
there is some pushback on fixing them. Joe Perches suggested that this
be hidden behind W=3 like all of the other signed warnings. This is also
disabled in Chrome OS's 4.14 branch.

Link: https://lore.kernel.org/lkml/20180619192504.180479-1-mka@chromium.org/
Suggested-by: Joe Perches <joe@perches.com>
Signed-off-by: Nathan Chancellor <natechancellor@gmail.com>
---
 scripts/Makefile.extrawarn | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/scripts/Makefile.extrawarn b/scripts/Makefile.extrawarn
index cf6cd0ef6975..2a21755f37bb 100644
--- a/scripts/Makefile.extrawarn
+++ b/scripts/Makefile.extrawarn
@@ -11,6 +11,7 @@
 # are not supported by all versions of the compiler
 # ==========================================================================
 
+KBUILD_CFLAGS += $(call cc-disable-warning, constant-conversion)
 KBUILD_CFLAGS += $(call cc-disable-warning, packed-not-aligned)
 
 ifeq ("$(origin W)", "command line")
@@ -47,6 +48,7 @@ warning-2 += $(call cc-option, -Wunused-macros)
 warning-3 := -Wbad-function-cast
 warning-3 += -Wcast-qual
 warning-3 += -Wconversion
+warning-3 += $(call cc-option, -Wconstant-conversion)
 warning-3 += -Wpacked
 warning-3 += -Wpadded
 warning-3 += -Wpointer-arith
-- 
2.19.1

